### PR TITLE
OCPBUGS-21584: UPSTREAM: 121128: [CVE-2023-39325] .: bump golang.org/x/net to v0.17.0

### PR DESCRIPTION
**Pull request by ncdc**
/kind bug

#### What this PR does / why we need it:
Bumping golang.org/x/net in light of CVE-2023-39325 and CVE-2023-44487.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Address CVE-2023-44487 and CVE-2023-39325 for all components that serve HTTP/2.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/api/go.mod [hunk 1]
```diff
@@ -38,16 +37,7 @@ require (
 )
 
 replace (
-	github.com/google/cadvisor => github.com/openshift/google-cadvisor v0.47.3-openshift-4.15-1
```
Comment (deads2k) [inline staging/src/k8s.io/api/go.mod:41]: This shouldn't have happened.  I'm going to try to pull locally and see if I get a different result.

**Comment by deads2k** [comment]
the go.mod doesn't look quite right here.  See if https://github.com/openshift/kubernetes/pull/1763 looks closer to the upstream and passes verify.

**Comment by soltysh** [comment]
/override ci/prow/unit
this is being tracked elsewhere, and the failures are not related to this change

**Comment by ncdc** [comment]
e2e-gcp transient failure - error building rpms
```
[ERROR] openshift-hack/build-rpms.sh:34: `OS_RPM_NAME="$( rpmspec -q --qf '%{name}\n' "${OS_RPM_SPECFILE}" | head -1 )"` exited with status 141.
```

/test e2e-gcp

**Comment by ncdc** [comment]
aws-ovn-serial failed due to quay outage
/test e2e-aws-ovn-serial

**Comment by deads2k** [comment]
/override ci/prow/e2e-gcp-ovn-upgrade

multiple observations
1. for TRT, the first unreadiness of master-2 is not listed on the timeline (15:03)
2. for networking, micro (already ovn-ic) upgrade had a networking outage for the master-2 host during the networking upgrade state.  This rippled into master-2 going ready=false, which impacts pod scheduling
3. for networking or TRT (networking preferred), let's get the pod sandbox creation failures posted to the timeline for future debugging
upshot for your PR: passed enough

**Comment by dgrisonnet** [comment]
@ncdc @deads2k we might have problems carrying the changes made in https://github.com/openshift/kubernetes/pull/1757/commits/d9d763d7410b30825a21b346503d687842b1843e since it wasn't marked as a <carry> in the commit.

**Comment by ncdc** [comment]
@dgrisonnet it's marked `UPSTREAM`

**Comment by dgrisonnet** [comment]
Yes but the part where we change UnauthenticatedHTTP2DOSMitigation to be default=true, should be a separate commit with UPSTREAM <carry> to make it easier to distinguish from the initial upstream commit.

**Comment by ncdc** [comment]
It is separate. The commit the changes the default to false was intentionally not cherry-picked, so we end up with the default of true. But I see your point. I guess a better approach would have been to pick the commit, then revert it with a `<carry>`.

**Comment by dgrisonnet** [comment]
Yeah we could've done it that way. When I made the patch in our k8s.io/apiserver fork, I treated both commits as one PR and squashed them together under UPSTREAM: 121120 and then added a <carry> patch: https://github.com/openshift/kubernetes-apiserver/pull/46

**Pull request merged**